pkg/river/token/builder/builder.go

```diff
@@ -166,9 +171,21 @@ func (b *Body) encodeFields(rv reflect.Value) {
 	}
 
 	fields := rivertags.Get(rv.Type())
+	clone := reflect.New(rv.Type()).Elem()
+
+	var goRiverDefaulter = reflect.TypeOf((*value.Defaulter)(nil)).Elem()
```

Comment:
Can this be a package-level variable so we're not always calculating it at runtime for each call?